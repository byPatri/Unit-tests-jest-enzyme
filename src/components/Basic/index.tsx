import React from 'react';

const Basic = () => (
    <div>Basic React Component</div>
);

export default Basic;
