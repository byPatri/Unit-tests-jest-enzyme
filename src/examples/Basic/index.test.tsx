import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Basic from './index';

describe('components/Basic/index', () => {
    it('should not throw on mount', () => {
        expect(() => mount(<Basic />)).not.toThrow();
    });

    it('should render correctly', () => {
        const wrapper = mount(<Basic />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should has proper text inside', () => {
        const wrapper = mount(<Basic />);
        const div = wrapper.find('div').at(0);

        expect(div.text()).toEqual('Basic React Component');
    });

    it('should has passed in props className', () => {
        const wrapper = mount(<Basic className='test-classname' />);
        const div = wrapper.find('div').at(0);

        expect(div.hasClass('test-classname')).toBe(true);
    });
});
