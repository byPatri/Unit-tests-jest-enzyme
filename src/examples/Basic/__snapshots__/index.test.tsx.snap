// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Basic/index should render correctly 1`] = `
<Basic>
  <div>
    Basic React Component
  </div>
</Basic>
`;
